src/publishPackages/publish.proj

```diff
@@ -53,33 +56,87 @@
 
   <Target Name="PreparePublish" DependsOnTargets="$(CreateDependsOn)">
     <MakeDir Directories="$(ManifestDir)" />
+    <MakeDir Directories="$(ArtifactsTempWorkDir)" />
     <PropertyGroup>
       <ExpectedFeedUrl>https://$(AzureAccountName).blob.core.windows.net/$(ContainerName)/index.json</ExpectedFeedUrl>
-      <AccountKey>$(AzureAccessToken)</AccountKey>
     </PropertyGroup>
+    <!-- Include other build data attributes : (Src) https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj#L167 -->
+    <ItemGroup>
+      <NewManifestBuildData Include="AzureDevOpsBuildId=$(BUILD_BUILDID)" />
+      <NewManifestBuildData Include="AzureDevOpsBuildDefinitionId=$(SYSTEM_DEFINITIONID)" />
+      <NewManifestBuildData Include="AzureDevOpsAccount=$(AzureAccountName)" />
+      <NewManifestBuildData Include="AzureDevOpsProject=$(SYSTEM_TEAMPROJECT)" />
+      <NewManifestBuildData Include="AzureDevOpsBuildNumber=$(ManifestBuildId)" />
+      <NewManifestBuildData Include="AzureDevOpsRepository=$(ManifestName)" />
+      <NewManifestBuildData Include="AzureDevOpsBranch=$(ManifestBranch)" />
+      <NewManifestBuildData Include="Location=$(ExpectedFeedUrl)" />
+      <!-- 
+        Adding this just to not fail the validation when creating the manifest. 
+        TODO: Should be removed once PublishingUsingPipelines is the default.
+      -->
+      <NewManifestBuildData Include="Location=$(AzureFeedUrl)" />
+    </ItemGroup>
   </Target>
 
-  <Target Name="PreparePublishPackages" DependsOnTargets="PreparePublish" BeforeTargets="PublishPackagesToBlobFeed">
+  <Target Name="PreparePublishPackages" DependsOnTargets="PreparePublish">
     <PropertyGroup>
-      <AssetManifestFilePath>$(ManifestDir)reference-packages-nupkg-manifest.xml</AssetManifestFilePath>
+      <AssetManifestFilename>reference-packages-nupkg-manifest.xml</AssetManifestFilename>
+      <AssetManifestFilePath>$(ManifestDir)$(AssetManifestFileName)</AssetManifestFilePath>
+      <!-- Create a temporary directory to store the generated asset manifest by the task -->
+      <TempWorkingDirectory>$(ArtifactsDir)AssetsTmpDir\$([System.Guid]::NewGuid())/</TempWorkingDirectory>
     </PropertyGroup>
     <ItemGroup>
       <PackagesToPublish Include="$(ArtifactsPkgDir)**/*.nupkg" />
     </ItemGroup>
+    <MakeDir Directories="$(TempWorkingDirectory)"/>
+    <PushToAzureDevOpsArtifacts
+      ItemsToPush="@(PackagesToPublish)"
+      ManifestBuildData="@(NewManifestBuildData)"
+      ManifestRepoUri="$(ManifestName)"
+      ManifestBranch="$(ManifestBranch)"
+      ManifestBuildId="$(ManifestBuildId)"
+      ManifestCommit="$(ManifestCommit)"
+      IsStableBuild="false"
+      AssetManifestPath="$(AssetManifestFilePath)"
+      AssetsTemporaryDirectory="$(TempWorkingDirectory)" />
+
+    <Message
```

Comment:
The `PushToAzureDevOpsArtifacts` task currently does the this step for you: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToAzureDevOpsArtifacts.cs#L149

Reply:
Sure thanks for the catch. I would also recommend a change in docs -> https://github.com/JohnTortugo/arcade/blob/9b3f304c7bc9fd4d11be9ca0b466b83e98d2a191/Documentation/CorePackages/Publishing.md#moving-away-from-the-legacy-pushtoblobfeed-task where the `becomes` part could confuse the reader to introduce the implicit duplication of vso stdout.